crates/rover-error/src/code.rs

```diff
@@ -0,0 +1,20 @@
+use houston::HoustonProblem;
+use std::convert::{TryFrom, TryInto};
+use std::fmt::{self, Display};
+
+#[derive(Debug)]
+pub enum Code {}
+
+impl Display for Code {
+    fn fmt(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(formatter, "no codes yet...")
+    }
+}
+
+impl TryFrom<&str> for Code {
+    type Error = &'static str;
+    fn try_from(type_name: &str) -> Result<Self, Self::Error> {
+        println!("ERrreRROORORR:\n{:?}", &type_name);
+        Err("Could not find a solution.")
+    }
+}
```

Comment:
this bit compiles, and maaayyy end up being what we have to do, but i'd like something a bit cleaner!

Reply:
So I am actually a lot more into this than what you have in `solution.rs`! I feel like if we ever run into a bug with status , we will have a much easier time debugging this than a three-layer generic imo. Additionally it's only for our own use and not like a public API, so we don't really have to make it super generic.

We could _also_ have, if you want, a static `into_status_code` and `as_status_code` for owned and borrowed values respectively. Static method for this kind of thing is good for debugging, but perhaps not as interesting as a `TryFrom` implementation.